`just fix` fails on `main` branch

## Description
running `just fix` fails on main 

output: 
```
error[E0308]: mismatched types
   --> crates/sqlexec/src/engine.rs:101:33
    |
101 | ...                   bucket,
    |                       ^^^^^^ expected `String`, found `Option<String>`
    |
    = note: expected struct `std::string::String`
                 found enum `std::option::Option<std::string::String>`

error[E0308]: mismatched types
   --> crates/sqlexec/src/engine.rs:183:43
    |
163 |         Ok(match (self.conf.clone(), session_conf.gcs_bucket.clone()) {
    |                  ---------------------------------------------------- this expression has type `(StorageConfig, std::option::Option<std::string::String>)`
...
183 |             (StorageConfig::Gcs { bucket: None, .. }, None) => {
    |                                           ^^^^ expected `String`, found `Option<_>`
    |
    = note: expected struct `std::string::String`
                 found enum `std::option::Option<_>`

error[E0308]: mismatched types
   --> crates/sqlexec/src/engine.rs:178:33
    |
178 |                         bucket: Some(bucket),
    |
    = note: expected struct `std::string::String`
                 found enum `std::option::Option<std::string::String>`

For more information about this error, try `rustc --explain E0308`.
error: could not compile `sqlexec` (lib) due to 3 previous errors
warning: build failed, waiting for other jobs to finish...
error[E0369]: binary operation `==` cannot be applied to type `StorageConfig`
   --> crates/sqlexec/src/engine.rs:470:9
    |
470 | /         assert_eq!(
471 | |             conf.conf,
...   |
478 | |             }
    | |_________|
    | |_________StorageConfig
    |           StorageConfig
    |
    = note: this error originates in the macro `assert_eq` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0369]: binary operation `==` cannot be applied to type `StorageConfig`
    |
484 |         assert_eq!(merged_conf.conf, StorageConfig::Memory,);
    |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    |         |
    |         StorageConfig
    |         StorageConfig
    |
    = note: this error originates in the macro `assert_eq` (in Nightly builds, run with -Z macro-backtrace for more info)

Some errors have detailed explanations: E0308, E0369.
```

--- a/.github/workflows/benchmarks.yaml
+++ b/.github/workflows/benchmarks.yaml
@@ -27,8 +27,6 @@ jobs:
           # Required to generate TPC-H data
           submodules: 'recursive'
 
-      - uses: dtolnay/rust-toolchain@1.73.0
-
       - uses: extractions/setup-just@v1
         name: Setup just
 

--- a/.github/workflows/ci.yaml
+++ b/.github/workflows/ci.yaml
@@ -18,10 +18,6 @@ jobs:
     steps:
       - name: Checkout
         uses: actions/checkout@v4
-      - name: Rust toolchain
-        uses: dtolnay/rust-toolchain@1.73.0
-        with:
-          components: clippy,rustfmt
       - uses: extractions/setup-just@v1
       - uses: actions/cache@v3
         name: Cache
@@ -41,10 +37,6 @@ jobs:
     steps:
       - name: Checkout
         uses: actions/checkout@v4
-      - name: Rust toolchain
-        uses: dtolnay/rust-toolchain@1.73.0
-        with:
-          components: clippy,rustfmt
       - uses: extractions/setup-just@v1
       - uses: actions/cache@v3
         name: Cache
@@ -67,10 +59,6 @@ jobs:
     steps:
       - name: Checkout
         uses: actions/checkout@v4
-      - name: Rust toolchain
-        uses: dtolnay/rust-toolchain@1.73.0
-        with:
-          components: clippy,rustfmt
       - uses: extractions/setup-just@v1
       - uses: actions/cache@v3
         name: Cache
@@ -98,10 +86,6 @@ jobs:
     steps:
       - name: Checkout
         uses: actions/checkout@v4
-      - name: Rust toolchain
-        uses: dtolnay/rust-toolchain@1.73.0
-        with:
-          components: clippy,rustfmt
       - uses: extractions/setup-just@v1
       - uses: actions/cache@v3
         name: Cache
@@ -124,10 +108,6 @@ jobs:
     steps:
       - name: Checkout
         uses: actions/checkout@v4
-      - name: Rust toolchain
-        uses: dtolnay/rust-toolchain@1.73.0
-        with:
-          components: clippy,rustfmt
       - uses: extractions/setup-just@v1
       - uses: actions/cache@v3
         name: Cache
@@ -136,9 +116,6 @@ jobs:
             ~/.cargo/
             target/
           key: ${{ runner.os }}-cargo-${{ hashFiles('**/Cargo.lock') }}
-      - name: Rust toolchain
-        uses: dtolnay/rust-toolchain@1.73.0
-
       - name: PG Protocol Tests
         run: |
           PROTOC=`just protoc && just --evaluate PROTOC` ./scripts/protocol-test.sh
@@ -153,10 +130,6 @@ jobs:
     steps:
       - name: Checkout
         uses: actions/checkout@v4
-      - name: Rust toolchain
-        uses: dtolnay/rust-toolchain@1.73.0
-        with:
-          components: clippy,rustfmt
       - uses: extractions/setup-just@v1
       - uses: actions/cache@v3
         name: Cache
@@ -177,10 +150,6 @@ jobs:
     steps:
       - name: Checkout
         uses: actions/checkout@v4
-      - name: Rust toolchain
-        uses: dtolnay/rust-toolchain@1.73.0
-        with:
-          components: clippy,rustfmt
       - uses: extractions/setup-just@v1
       - uses: actions/cache@v3
         name: Cache

--- a/.github/workflows/python-release.yaml
+++ b/.github/workflows/python-release.yaml
@@ -23,7 +23,6 @@ jobs:
         target: [x86_64]
     steps:
       - uses: actions/checkout@v4
-      - uses: dtolnay/rust-toolchain@1.73.0
       - uses: extractions/setup-just@v1
       - name: install protoc
         run: just protoc
@@ -52,7 +51,6 @@ jobs:
         target: [x64]
     steps:
       - uses: actions/checkout@v4
-      - uses: dtolnay/rust-toolchain@1.73.0
       - uses: extractions/setup-just@v1
       - name: install protoc
         run: just protoc
@@ -84,7 +82,6 @@ jobs:
         target: [x86_64, aarch64]
     steps:
       - uses: actions/checkout@v4
-      - uses: dtolnay/rust-toolchain@1.73.0
       - uses: extractions/setup-just@v1
       - name: install protoc
         run: just protoc

--- a/rust-toolchain.toml
+++ b/rust-toolchain.toml
@@ -1,0 +1,2 @@
+[toolchain]
+channel = "1.73"

--- a/rust-toolchain.toml
+++ b/rust-toolchain.toml
@@ -1,2 +1,3 @@
 [toolchain]
-channel = "1.73"
+channel = "1.73.0"
+components = ["clippy", "rustfmt"]

